goncy/twincy
# [WIP]: Replace Electron with Tauri

## Description
The goal of this PR is to migrate away from Electron to [Tauri](https://tauri.studio/)

## Instructions to test the solution
- Make sure you have followed the instructions to [install the required tools](https://tauri.studio/docs/getting-started/prerequisites) to use Tauri in your OS
- Run the app in dev mode following the steps in the updated README
- In my testing, to capture the client window in OBS we need to:
  - Add Window Capture Source
  - Select `Windows 10 (1903 and up)` capture method
  - Maximize the client window
  - Add `Color key` filter (I had to use green as the key color and update the background color in the client to match to ensure the selected message looks good)

## Notes
- The client window it's being opened by default for now. Maybe we could change the `Copy` button in the admin to open the client instead of just giving you the URL
- For now the resulting app from the build command doesn't work properly. This is caused by the app not including the server and client code by default. As discussed in discord we need to decide how to implement the solution taking into account the desired DX and other factors.
